src/coreclr/src/tools/aot/crossgen2/CommandLineOptions.cs

```diff
@@ -1,6 +1,7 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+#if SHOULD_RUN_CROSSGEN
```

Comment:
Why do this change need to be touching crossgen2? I would expect that we will run the linker analysis for libraries only.

Reply:
I was able to avoid the warnings (samples shown below) I was seeing locally relating to crossgen/crossgen2 and other projects changed in this PR by enabling verbose output on only libraries (i.e not mono/corelib System.Private.CoreLib). These warnings motivated most of the changes in this PR.

```
CROSSGEN-CORELIB : error : CrossGen System.Private.CoreLib build failed. Refer to D:\repos\dotnet_runtime\src\coreclr\..\..\artifacts\log\CrossgenCoreLib_Windows_NT__x64__Release.log [D:\repos\dotnet_runtime\src\coreclr\crossgen-corelib.proj]
EXEC : error : file "D:\repos\dotnet_runtime\src\coreclr\..\..\artifacts\bin\coreclr\Windows_NT.x64.Release\IL\System.Private.CoreLib.dll" or one of its dependencies was not found [D:\repos\dotnet_runtime\src\coreclr\crossgen-corelib.proj]
EXEC : error : Error enumerating files under D:\repos\dotnet_runtime\src\coreclr\..\..\artifacts\bin\coreclr\Windows_NT.x64.Release\IL\. [D:\repos\dotnet_runtime\src\coreclr\crossgen-corelib.proj]
D:\repos\dotnet_runtime\src\coreclr\crossgen-corelib.proj(20,5): error MSB3073: The command ""D:\repos\dotnet_runtime\src\coreclr\crossgen-corelib.cmd" -x64 -release" exited with code 1.

D:\repos\dotnet_runtime\.dotnet\sdk\5.0.100-preview.8.20362.3\Microsoft.Common.CurrentVersion.targets(4651,5): error MSB3030: Could not copy the file "D:\repos\dotnet_runtime\artifacts\bin\coreclr\OSX.x64.Release\clrjit.dll" because it was not found. [D:\repos\dotnet_runtime\src\coreclr\src\tools\aot\crossgen2\crossgen2.csproj]
D:\repos\dotnet_runtime\.dotnet\sdk\5.0.100-preview.8.20362.3\Microsoft.Common.CurrentVersion.targets(4651,5): error MSB3030: Could not copy the file "D:\repos\dotnet_runtime\artifacts\bin\coreclr\OSX.x64.Release\jitinterface.dll" because it was not found. [D:\repos\dotnet_runtime\src\coreclr\src\tools\aot\crossgen2\crossgen2.csproj]
```